Fix splitting of long messages

MadLittleMods opened the pull request:
Based off of @dequis PR https://github.com/gitterHQ/irc-bridge/pull/61

 - Fixes issue where part of a message is missing if using colons and is a long message.
 - Will split out long messages without spaces into proper 512 chunks (according to IRC spec)

### Todo:

 - [x] Change `MESSAGE_PIECE_MAX_LENGTH` to 450 so that there is a 60 character margin and words don't get cut-off. Thanks @dequis, https://github.com/gitterHQ/irc-bridge/pull/61#discussion_r67290804

---

 - Tested in HexChat

MadLittleMods commented:
@dequis Can you take a look at this PR which iterates on what you have done?

dequis commented:
This does look much cleaner, but it no longer tries to split words by space. This might be acceptable. Up to you to decide.

The tests look verbose as hell and that's a Good Thing. Certainly easier to understand than my version.

So, if splitting only at the length limit and not at word boundaries is okay, this looks good to me.

(I forgot to submit this comment before and it's a bit out of date with the TODO edited in the description, but, eh, whatever)

MadLittleMods commented:
@dequis I updated the chunking function and added some tests around it. Would appreciate another review pass and any potential test cases I am missing.
